### PR TITLE
First implementation of Room Persistency

### PR DESCRIPTION
**Pull request by pfent**
## Issue
see #560 

Currently working around our legacy database querying on the main thread with `allowMainThreadQueries()`.
We probably need to change this sometime in the future, since that's really not the best way to do it

Also: We kinda need to think about how we want to handle migrations in the future, deleting everything was 
never a good idea in the first place...

### CODE REVIEW

**Comment by kordianbruck** [comment]
@pfent looks good. I would leave this open, till the students from the current practical course take a look at it :100:

**Comment by pfent** [comment]
@anselm92 Could you already take look at it?

**Comment by anselm92** [comment]
Sorry for the delay, missed the email.

I had a look 👍 So what still has to be done is:

- Move queries from mainthread to seperate thread
- Perhaps add some tests
- Migration strategy

Am i right?

**Comment by kordianbruck** [comment]
@anselm92 plus add implementation for all the other db tables :1st_place_medal:

**Comment by pfent** [comment]
@anselm92 I just added a checklist in https://github.com/TCA-Team/TumCampusApp/issues/560#issuecomment-348933857

File: build.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ allprojects {
         options.forkOptions.setMemoryMaximumSize("4g") // Set maximum memory to 4g
         options.compilerArgs << "-Xlint:all"
         options.compilerArgs << "-Xlint:-classfile"
-        options.compilerArgs << "-Werror"
+        //options.compilerArgs << "-Werror"
```
Comment (kordianbruck) [inline build.gradle]: Nice!

**Comment by kordianbruck** [comment]
@anselm92 in the future please actually do a code review before approving a pull request. Thanks.

**Comment by anselm92** [comment]
Actually I did.

**Comment by kordianbruck** [comment]
@anselm92 alright - sorry, it rarely happens, that one does not find anything to comment on a pull request :grinning:

**Pull request merged**